File: Python/labs/lab4/task_2_2/WeatherWarnings.py
from City import City


class WeatherWarnings:
    MAX_TEMPERATURE = 100
    MIN_TEMPERATURE = 16

    def postWarning(self, city: City):
        if (city.getTemperature() >= WeatherWarnings.MAX_TEMPERATURE
                or city.getTemperature() <= WeatherWarnings.MIN_TEMPERATURE):
            print(f"Warning! Current temperature in {city.getName()} is "
                  f"{city.getTemperature()} {city.getTemperatureScale()}")
            city.setHasWeatherWarning(True)
        else:
            print("Temperature in " + city.getName() + " is OK.")

File: Cpp/lab07_Adapter/task_2_1/src/weather_warnings.cpp
#include <iostream>

#include "weather_warnings.hpp"


WeatherWarnings::WeatherWarnings() {}

void WeatherWarnings::postWarning(City& city) const
{
  if (city.getTemperature() >= WeatherWarnings::MAX_TEMPERATURE || city.getTemperature() <= WeatherWarnings::MIN_TEMPERATURE)
  {
    std::cout << "Warning! Current temperature in " << city.getName() << " is " << city.getTemperature()
    << " " << city.getTemperatureScale() << std::endl;

    city.setHasWeatherWarning(true);
  }
  else
  {
    std::cout << "Temperature in " << city.getName() << " is OK." << std::endl;
  }
}
